### PR TITLE
#434, Add ability to sort boards by code/name in ascending/descending order

### LINKED ISSUE #434: Feature request: Sorting mode for boards
Please add a button or option to sort boards by their name instead of just their code.

### PR DESCRIPTION
**Pull request by K1rakishou**
Closes #434

### CODE REVIEW

File: Clover/app/src/main/java/org/floens/chan/core/presenter/BoardSetupPresenter.java [hunk 1]
```diff
@@ -315,4 +347,120 @@ public long getId() {
...
+
+        private int options;
+
+        public BoardSuggestionComparator(BoardSortingInfo boardSortingInfo) {
+            options = boardSortingInfo.getSortingMode().value | boardSortingInfo.getSortingOrder().value;
+        }
+
+        @Override
+        public int compare(BoardSuggestion o1, BoardSuggestion o2) {
+            if (o1 == null && o2 == null) {
```
Comment (K1rakishou) [inline Clover/app/src/main/java/org/floens/chan/core/presenter/BoardSetupPresenter.java]: Probably should just throw an exception when either of parameters is null.

**Comment by Adamantcheese** [comment]
I think what was requested (which is also #703) is to sort SAVED boards, not boards that are being added. This would completely override the existing fuzzy search suggestion method.

**Comment by K1rakishou** [comment]
>There was a sort A-Z option in the board editor, we should bring it back.

Isn't board editor the window where you add boards?

**Comment by Adamantcheese** [comment]
Yeah, but with your current implementation the fuzzy search is completely ignored with no way to turn it back on. And either way there's no real reason to have it being sorted in any order with a search bar, just find what you want that way.

**Comment by K1rakishou** [comment]
@Adamantcheese 
>And either way there's no real reason to have it being sorted in any order with a search bar, just find what you want that way.

Yeah, I guess it makes sense. But then I don't get what boards need to be sorted. What window exactly is it.

**Comment by Adamantcheese** [comment]
I don't know either. People aren't being terribly specific with what boards they want sorted. I assumed that it was saved boards because that's what you look at the most?

**Comment by K1rakishou** [comment]
By saved boards you mean this spinner?

**Comment by Adamantcheese** [comment]
Yeah

**Comment by K1rakishou** [comment]
Well, I guess this task can wait until the next release. I asked to elaborate on what exactly needs to be sorted.

**Pull request closed without merging**